dokka-runners/dokka-gradle-plugin/src/main/kotlin/internal/ExperimentalFlags.kt

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright 2014-2024 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license.
+ */
+package org.jetbrains.dokka.gradle.internal
+
+import org.gradle.api.Project
+import org.gradle.kotlin.dsl.extra
+
+internal data class ExperimentalFlags(
+    val dokkatooEnabled: Boolean,
+    val dokkatooEnabledQuietly: Boolean,
+) {
+
+    companion object {
+        @Suppress("ConstPropertyName")
+        private const val PropName = "DokkaGradlePluginMode"
+
+        fun Project.dokkaMode(): ExperimentalFlags {
+
+            val flags = providers
+                .gradleProperty(PropName)
+                .forUseAtConfigurationTimeCompat()
+                .orElse(
+                    project.provider { project.extra.properties[PropName]?.toString() ?: "" }
+                )
+                .orNull
+                ?.split(",")
+                .orEmpty()
+                .map { it.lowercase() }
+                .toSet()
+
+            val dokkatooEnabledQuietly = "dokkatoo!" in flags
+            val dokkatooEnabled = dokkatooEnabledQuietly || "dokkatoo" in flags
```

Comment:
Here's the flag that will toggle between DGP-classic and DGP-nu.

i.e. A user can add `DokkaGradlePluginMode=dokkatoo` to switch to the new plugin mode.

Feel free to bike-shed about the property name!

Reply:
Would be nice to have something like this `org.jetbrains.dokka.experimental.useNewDokkaGradlePluginMode=true` , naming is hard, but would be nice to have at least the same prefix as in `org.jetbrains.dokka.experimental.tryK2`. Not sure why we need `quietly` as there are only 2 states:
1. classic plugin - `warning` log will be emitted
2. new (dokkatoo) plugin - no `warning` log will be emitted :)

Of course, it's fine to change it later, just let's not forget about it

Reply:
Yes, renaming to `org.jetbrains.dokka.experimental.useNewDokkaGradlePluginMode` makes sense. Although I very much prefer enums over boolean flags in public API, because they are more flexible, descriptive, and fine-grained.

What about something like `org.jetbrains.dokka.experimental.tryGradlePlugin=v2`?

Reply:
I would argue that we need any more flexibility here, because we really need just a `flag`: use new or old plugin.
As a user, in this case, I would read `v2` as a separate `version` field which could change in future once more.
Let's not give hints, that we would have `v3` :)

Also, boolean flag is consistent with what we have now with `tryK2` and most flags in KGP also follow the same idea: if there is just switch to new shiny behaviour or some experimental thing - it's just a flag.

Though, `new` in naming looks bad, as it's hard to understand when it becomes `old`.

Overall, I'm fine with any variant which has `org.jetbrains.dokka.experimental` prefix.

Reply:
I added the 'quietly' function because there's also a logged message for if DGP is in 'dokkatoo' mode, which instantly annoyed me! But I think the 'no warning' option is still relevant.

The plan is to have a phased rollout for Dokkatoo:

1. Release the merged DokkaClassic/Dokkatoo plugin.
   DokkaClassic is the default mode.
   There's no warning for DokkaClassic.
   Users can opt-in to Dokkatoo, with a logged message with a link to the migration docs and 'create new GitHub issue'.
3. Dokkatoo is stable, and is made the default.
   Deprecate DokkaClassic.
   Users can revert to DokkaClassic, with a logged warning (with a link to the migration docs).
4. Remove DokkaClassic, remove the opt-in flag.

In steps 1 and 2 it would be nice to be able to suppress the warnings.

Reply:
okay, agree, that quietly option is needed, let's stick then with what is in this PR and just rename property to `org.jetbrains.dokka.experimental.gradlePluginMode`

Reply:
> Though, `new` in naming looks bad, as it's hard to understand when it becomes `old`.

Naming things as 'NewFoo' is a pet peeve of mine :)

Reply:
I've updated the properties and the messages in a new PR

https://github.com/Kotlin/dokka/pull/3736